src/Data/Food/Retail.elm

```diff
@@ -0,0 +1,216 @@
+module Data.Food.Retail exposing
+    ( Conservation
+    , all
+    , ambient
+    , computeImpacts
+    , decode
+    , displayNeeds
+    , encode
+    , fromString
+    , toDisplay
+    , toString
+    )
+
+{- This module allow to compute the impacts of the transport of finished products to the retail stores,
+   and the impact of storing the product at the store
+-}
+
+import Data.Food.Builder.Db exposing (Db)
+import Data.Food.Process exposing (Process, WellKnown)
+import Data.Impact as Impact exposing (Impacts)
+import Data.Unit as Unit
+import Energy exposing (Joules, kilowattHours)
+import Json.Decode as Decode exposing (Decoder)
+import Json.Encode as Encode
+import Length exposing (Length)
+import Mass exposing (Mass)
+import Quantity exposing (Quantity, Rate, rate, ratio)
+import Result.Extra as RE
+import Volume exposing (CubicMeters, Volume, cubicMeters, liters)
+
+
+type
+    Conservation
+    -- A consevation type and its needs in energy, cooling, water and transport
+    = Conservation Type Needs
+
+
+type Type
+    = Ambient
+    | Fresh
+    | Frozen
+
+
+type alias Needs =
+    --- what it needs to store a product at the retail store
+    { energy : Quantity Float (Rate Joules CubicMeters)
+    , cooling : Quantity Float (Rate Joules CubicMeters)
+    , water : Float
+    , transport : Length
+    }
+
+
+
+-- Data table from https://fabrique-numerique.gitbook.io/ecobalyse/alimentaire/etapes-du-cycles-de-vie/vente-au-detail
+
+
+ambient : Conservation
+ambient =
+    Conservation Ambient
+        { energy = rate (kilowattHours 123.08) (cubicMeters 1)
+        , cooling = rate (kilowattHours 0) (cubicMeters 1)
+        , water = ratio (liters 561.5) (cubicMeters 1)
+        , transport = Length.kilometers 600
+        }
+
+
+fresh : Conservation
+fresh =
+    Conservation Fresh
+        { energy = rate (kilowattHours 46.15) (cubicMeters 1)
+        , cooling = rate (kilowattHours 219.23) (cubicMeters 1)
+        , water = ratio (liters 210.6) (cubicMeters 1)
+        , transport = Length.kilometers 600
+        }
+
+
+frozen : Conservation
+frozen =
+    Conservation Frozen
+        { energy = rate (kilowattHours 61.54) (cubicMeters 1)
+        , cooling = rate (kilowattHours 415.38) (cubicMeters 1)
+        , water = ratio (liters 280.8) (cubicMeters 1)
+        , transport = Length.kilometers 600
+        }
+
+
+displayNeeds : Conservation -> String
+displayNeeds (Conservation type_ needs) =
+    let
+        energy =
+            needs.energy |> Quantity.in_ (Energy.kilowattHours >> Quantity.per (Volume.cubicMeters 1)) |> String.fromFloat
+
+        cooling =
+            needs.cooling |> Quantity.in_ (Energy.kilowattHours >> Quantity.per (Volume.cubicMeters 1)) |> String.fromFloat
+
+        water =
+            needs.water / ratio (Volume.liters 1) (Volume.cubicMeters 1) |> String.fromFloat
+
+        transport =
+            needs.transport |> Length.inKilometers |> String.fromFloat
+    in
+    case type_ of
```

Comment:
Am I missing a difference between the three types? The only difference I can spot is the typo in `Fresh` for `Transpor` with is missing `t: `.

I guess we could drop the `case` if all three branches are exactly the same?

Reply:
Of course, thx !